src/lyra/abstract_domains/state.py

```diff
@@ -142,6 +142,25 @@ def output(self, output: Set[Expression]) -> 'State':
         self.result = set()  # outputs have no result, only side-effects
         return self
 
+    @abstractmethod
+    def _raise_error(self) -> 'State':
+        """Outputs something in the current state.
+
+        :param output: expression representing the output
+        :return: current state modified by the output
+
+        """
+
+    def raise_error(self) -> 'State':
+        """Raises an error
+
+        :return: current state modified by the error
+
+        """
+        self._raise_error()
+        self.result = set()  # outputs have no result, only side-effects
+        return self
```

Comment:
No need for both 'raise_error' and '_raise_error' here. Just use 'raise_error' and leave it abstract.